packages/stencil-library/src/components/button/button.tsx

```diff
@@ -25,57 +21,87 @@ export class Button {
                     return "usa-button--outline"
                 case 'link':
                     return "usa-button--unstyled"
-                case 'danger':
-                    return "usa-button--secondary"
             }
         }
 
         const getDarkModeVariantClassName = (variant: ButtonVariant) => {
             switch (variant) {
+                case 'primary':
+                    return "primary-button-dark"
                 case "secondary":
                     return "usa-button--outline usa-button--inverse"
                 case 'link':
-                    return "usa-button--unstyled usa-button--outline usa-button--inverse"
+                    return "usa-button--unstyled unstyled-button-dark"
+                default:
+                    return 'primary-button-dark'
+            }
+        }
+
+        const getDangerModeVariantClassName = (variant: ButtonVariant) => {
+            switch (variant) {
+                case 'primary':
+                    return "usa-button--secondary"
+                case "secondary":
+                    return "usa-button--outline outline-danger"
+                case 'link':
+                    return "usa-button--unstyled unstyled-button-danger"
                 default:
                     return getVariantClassName(variant)
             }
         }
 
-        return this.mode === "light"
-            ? `usa-button ${getVariantClassName(this.variant)}`
-            : `usa-button ${getDarkModeVariantClassName(this.variant)}`
+        let getClassName;
```

Comment:
[Dust] IMO setting the value of the classname (by invoking the function) rather than setting the function to call may be clearer for a dev